Automattic/wp-calypso
# Update dataviews package version to latest

Fixes https://github.com/Automattic/jetpack-manage/issues/337

## Proposed Changes

* Update dataviews package version.
* Remove locks on the version changes.
* Downstream changes to use the new package version on the Hosting Dashboard (/sites), the A4A Dashboard, and the Add Site > Via WordPress.com modal.

Before | After
---- | -----
<img width="667" alt="Screen Shot 2024-09-03 at 11 32 16 AM" src="https://github.com/user-attachments/assets/e9bfb423-9de8-4371-aa44-975b702d7d46"> | <img width="710" alt="Screen Shot 2024-09-03 at 11 31 39 AM" src="https://github.com/user-attachments/assets/5d424269-8dfb-4725-befa-eb85f7bd2fe0">


## Why are these changes being made?


* To update to the latest version of `DataViews` in Core. See pe7F0s-21J-p2

## Status

| Status | DataViews instance | URL |
| --- | --- | --- |
| 🟢    | [Dotcom Sites](https://github.com/Automattic/wp-calypso/blob/453b8b2e3cdf43e2a763725a835b0e6ea7aeced0/client/hosting/sites/components/sites-dataviews/index.tsx#L270) | `<calypso>/sites` |
| 🟢  | [A4A Sites](https://github.com/Automattic/wp-calypso/blob/453b8b2e3cdf43e2a763725a835b0e6ea7aeced0/client/a8c-for-agencies/sections/sites/features/jetpack/jetpack-sites-dataviews.tsx#L587) | `<agencies>/sites` |
| 🟢   | [A4A Sites Modal](https://github.com/Automattic/wp-calypso/blob/68d9d691892ef0c19295cea3fd30568c37858a83/client/a8c-for-agencies/components/add-new-site-button/import-from-wpcom-modal/table-content.tsx#L57) | `<agencies>/sites` ("add new site" modal) |
| 🟢  | [A4A Referrals Overview](https://github.com/Automattic/wp-calypso/blob/68d9d691892ef0c19295cea3fd30568c37858a83/client/a8c-for-agencies/sections/referrals/referrals-list/index.tsx#L223) | `<agencies>/referrals` |
| 🟢  | [A4A Referrals Details](https://github.com/Automattic/wp-calypso/blob/68d9d691892ef0c19295cea3fd30568c37858a83/client/a8c-for-agencies/sections/referrals/common/referral-details-table/index.tsx#L19) | `<agencies>/referrals` (with an item selected) |
| 🟢  | [A4A Referrals for clients](https://github.com/Automattic/wp-calypso/blob/trunk/client/a8c-for-agencies/sections/client/primary/subscriptions-list/index.tsx#L115) | `<agencies>/clients/subscriptions` | 
| 🟢 | [A4A Team](https://github.com/Automattic/wp-calypso/blob/68d9d691892ef0c19295cea3fd30568c37858a83/client/a8c-for-agencies/sections/team/primary/team-list/index.tsx#L174) (in development, [not in production](https://github.com/Automattic/wp-calypso/blob/trunk/config/a8c-for-agencies-production.json#L57)) | `<agencies>/team` |
| 🟢 | Jetpack Cloud Sites (legacy, [not in production](https://github.com/Automattic/wp-calypso/blob/trunk/config/jetpack-cloud-production.json#L89)): [one](https://github.com/Automattic/wp-calypso/blob/68d9d691892ef0c19295cea3fd30568c37858a83/client/a8c-for-agencies/sections/sites/sites-dataviews/index.tsx#L448), [two](https://github.com/Automattic/wp-calypso/blob/68d9d691892ef0c19295cea3fd30568c37858a83/client/jetpack-cloud/sections/agency-dashboard/sites-overview/sites-dataviews/index.tsx#L313). | `<jpcloud>/sites` |

Dotcom Sites

- Nothing to fix in this PR.

A4A Sites

- Row notice for sites whose status is "Backup failed".

A4A Sites Modal

- Nothing to fix in this PR.

A4A Referrals Overview

- Nothing to fix in this PR.

A4A Referrals Details

- Nothing to fix in this PR.

A4A Referrals for clients

- Infinite data loop. There's a fix but not sure if it's the right one ([conversation](https://github.com/Automattic/wp-calypso/pull/93503/commits/1c87b369395f86a43744208e02ec6f90dde9f18c#r1738985766)).
- Actions column name is not aligned to the right as it was before. This is because existing CSS targets `data-field-id="actions"` that no longer exists in DataViews.

A4A Team

- Nothing to fix in this PR.

Jetpack Cloud

- These are not in use, they should be removed from the codebase.

## Testing Instructions



* Check out this branch locally, or use the Calypso live links below.
* Open /sites. Verify that:
    * the list of sites loads and has pagination. 
    * you can click on a site and open the fly out panel, and the table collapses into a list. The sidebar collapses into a rail.
    * you can search and filter the sites.
    * the table loads only Site and Actions on mobile.
* Open the Agencies dashboard. Verify that:
    * the list of sites loads and has pagination.
    * you can click on a site and open the fly out panel, and the table collapses into a list.
    * you can search and filter the sites.
    * you can favorite sites.

## Pre-merge Checklist
